providers/test/Test.fst

```diff
@@ -359,126 +374,347 @@ let test_dh () : St unit =
   // TODO supposed to use tactics now?
   ()
 
-#reset-options "--z3rlimit 40"
+inline_for_extraction
+noeq
+type features = {
+  features_avx: bool;
+  features_avx2: bool;
+  features_bmi2: bool;
+  features_adx: bool;
+  features_aesni: bool;
+  features_shaext: bool;
+}
+
+inline_for_extraction
+let f_concat (f1 f2: features) : Tot features =
+  [@inline_let]
+  let avx = f1.features_avx || f2.features_avx in
+  [@inline_let]
+  let avx2 = f1.features_avx2 || f2.features_avx2 in
+  [@inline_let]
+  let bmi2 = f1.features_bmi2 || f2.features_bmi2 in
+  [@inline_let]
+  let adx = f1.features_adx || f2.features_adx in
+  [@inline_let]
+  let aesni = f1.features_aesni || f2.features_aesni in
+  [@inline_let]
+  let shaext = f1.features_shaext || f2.features_shaext in
+  {
+    features_avx = avx;
+    features_avx2 = avx2;
+    features_bmi2 = bmi2;
+    features_adx = adx;
+    features_aesni = aesni;
+    features_shaext = shaext;
+  }
+
+inline_for_extraction
+let f_none : features = {
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_avx : features = // [@inline_let] ({ f_none with features_avx = true; })
```

Comment:
why didn't this work? should we fix it?

Reply:
In fact, the `{x with field = value}` syntax makes F* add a `let` binding, which would have to be inlined for extraction. This is not the case currently. Ideally we should allow the user to ask for this introduced `let` binding to be inlined at extraction, through, say, `[@inline_let]`. Right now this is not done. So I think it is really on the F* syntax side.
(More generally, we should review all F* "syntactic sugar" constructs silently introducing `let` bindings.)